### PR TITLE
(PUP-8900) Use HTTP client to upload facts

### PR DESCRIPTION
**Pull request by joshcooper**
Previously, `puppet facts upload` only tried the first entry in `server_list`,
but never fell back to other entries in the list.

Now it uses the http client which attempts to connect to each entry in the
`server_list`, so that `puppet facts upload` behaves the same as `puppet
agent` for SRV, `server_list` and `server` resolution.

There's one minor change to the log message to remove the extra ':'

Faces will output to the console whatever the `when_invoked` block
returns, so return an explicit `nil` which is what `indirection.save`
did.

### CODE REVIEW

**Comment by joshcooper** [comment]
jenkins please test this with servertests on redhat7-64a,windows2019-64a

File: lib/puppet/face/facts.rb [hunk 1]
```diff
@@ -75,13 +75,16 @@
...
+      puppet = session.route_to(:puppet)
+
+      Puppet.notice(_("Uploading facts for '%{node}' to '%{server}'") % {
                     node: Puppet[:node_name_value],
-                    server: server})
+                    server: puppet.url.hostname})
 
-      Puppet::Node::Facts.indirection.save(facts, nil, :environment => Puppet.lookup(:current_environment))
+      puppet.put_facts(Puppet[:node_name_value], facts: facts, environment: Puppet.lookup(:current_environment).name.to_s)
+      nil
```
Comment (melissa) [inline lib/puppet/face/facts.rb]: Why are we returning `nil` here?

Comment (joshcooper) [reply lib/puppet/face/facts.rb]: ah, faces automatically prints out whatever the last line returns. Previously it was relying on `indirection.save` returning nil (I think). Let me check/confirm and update the commit message.

Comment (joshcooper) [reply lib/puppet/face/facts.rb]: Yeah it returns `nil`. I updated the commit message so it's clear why I added that.

**Comment by melissa** [comment]
I'll merge this once tests pass

**Pull request merged**